### ISSUE TITLE
New feature: allow titleRegex to retrieve end date of multiday events if that date is in their title

### ISSUE THREAD

**Issue by johannesnoordanus** [edwardspec/mediawiki-extension-JsCalendar#4]
It would be nice to be able to specify an event that is spanning several days without having to enter all days separately.
I implemented this by adding _enddate_ handling in _EventCalendar.php_ (see attached file).
It is now possible to (for example) specify 
> ```
> <EventCalendar>
> aspectratio = 2.0
> titleRegex = ^([0-9]{4,4}/[0-9][0-9]/[0-9][0-9])(:[0-9]{4,4}/[0-9][0-9]/[0-9][0-9])?_.*
> dateFormat = Y/m/d
> symbols = 200
> keywordcolor.afspraak = red
> </EventCalendar>
>```
 and match page titles like _2022/04/28:2022/04/29_testtest_ . 
This is backwards compatible with the previous regex and pages like _2022/04/29_testtest_ (without a colon sign) still match. 
Note that use of a colon ':' isn't strictly necessary and done because of readability.

Note: not a feature of this update, but if you start an event page by entering the start time followed by a description, this will be displayed within the specific calendar day.

I added the updated .php file to this issue.
[EventCalendar.zip](https://github.com/edwardspec/mediawiki-extension-JsCalendar/files/8564440/EventCalendar.zip)

**Comment by edwardspec** [comment]
Thank you for your contribution.

> EventCalendar.zip

Can you please send a pull request? (click "Fork" button in top-right corner, change the .php file in your repository, then click "Contribute" -> "Open pull request")

**Comment by johannesnoordanus** [comment]
> Thank you for your contribution.
> 
> > EventCalendar.zip
> 
> Can you please send a pull request? (click "Fork" button in top-right corner, change the .php file in your repository, then click "Contribute" -> "Open pull request")

As far as I can see, I created the pull request as you indicated.

**Comment by edwardspec** [comment]
I did some preliminary cleanup and added a unit test. It seems to work correctly.
Branch: https://github.com/edwardspec/mediawiki-extension-JsCalendar/commits/multiday-event-regex

Notable changes:
1) Having symbol `:` between start/end date is no longer required (it can be any other separator). This `:` will still be trimmed away if the event name starts/ends with `:`.
2) end date is no longer increased by +1 day. End date (explicitly specified by user) is the exact day when the event ends. Adding +1 was only needed in situations when we don't know the end date (in which case it's start date +1 day).
3) if end date is incorrect format, but start date is not, then event is not skipped. Instead, it's treated as 1 day event, as if only the start date was specified.

Concerns before this is merged:
1) I'm worried that some users might need to use braces, e.g. `(someText|anotherString)`, in their `titleRegex`, and it may become `$matches[2]`. This is why I changed the code to tolerate incorrect end date formats. This problem can probably be avoided by recommending the use of `(?:contents of these braces won't be matched)` syntax in documentation, and covering this with tests.
